cmd/searcher/protocol/searcher.go

```diff
@@ -58,6 +61,23 @@ type Request struct {
 	Indexed bool
 }
 
+func (p *Request) ValidateParams() error {
+	if p.Repo == "" {
+		return errors.New("Repo must be non-empty")
+	}
+	// Surprisingly this is the same sanity check used in the git source.
+	if len(p.Commit) != 40 {
+		return errors.Errorf("Commit must be resolved (Commit=%q)", p.Commit)
+	}
+	if p.Pattern == "" && p.ExcludePattern == "" && len(p.IncludePatterns) == 0 {
+		return errors.New("At least one of pattern and include/exclude pattners must be non-empty")
+	}
+	if p.IsNegated && p.IsStructuralPat {
+		return errors.New("Negated patterns are not supported for structural searches")
```

Comment:
```suggestion
		return errors.New("repo must be non-empty")
	}
	// Surprisingly this is the same sanity check used in the git source.
	if len(p.Commit) != 40 {
		return errors.Errorf("commit must be resolved (Commit=%q)", p.Commit)
	}
	if p.Pattern == "" && p.ExcludePattern == "" && len(p.IncludePatterns) == 0 {
		return errors.New("at least one of pattern and include/exclude pattners must be non-empty")
	}
	if p.IsNegated && p.IsStructuralPat {
		return errors.New("negated patterns are not supported for structural searches")
```

nit: error strings start with lowercase as the general Go convention :D